nexus/src/db/datastore/volume.rs

```diff
@@ -447,6 +467,195 @@ impl DataStore {
+
+                            // Update the original volume_id with the new
+                            // volume.data.
+                            use db::schema::volume::dsl as volume_dsl;
+                            let num_updated = diesel::update(volume_dsl::volume)
+                                .filter(volume_dsl::id.eq(volume_id))
+                                .set(volume_dsl::data.eq(new_volume_data))
+                                .execute(conn)?;
+
+                            // This should update just one row.  If it does
+                            // not, then something is terribly wrong in the
+                            // database.
+                            if num_updated != 1 {
+                                return Err(TxnError::CustomError(
+                                    RemoveReadOnlyParentError::UnexpectedDatabaseState,
+                                ));
+                            }
+
+                            // Make a new VCR, with the information from
+                            // our temp_volume_id, but the read_only_parent
+                            // from the original volume.
+                            let rop_vcr = VolumeConstructionRequest::Volume {
+                                id: temp_volume_id,
+                                block_size,
+                                sub_volumes: vec![],
+                                read_only_parent,
+                            };
+                            let rop_volume_data =
+                                serde_json::to_string(
+                                    &rop_vcr
+                                )
+                                .map_err(|e| {
+                                    TxnError::CustomError(
+                                        RemoveReadOnlyParentError::SerdeError(
+                                            e,
+                                        ),
+                                    )
+                                })?;
+                            // Update the temp_volume_id with the volume
+                            // data that contains the read_only_parent.
+                            let num_updated =
+                                diesel::update(volume_dsl::volume)
+                                    .filter(volume_dsl::id.eq(temp_volume_id))
+                                    .filter(volume_dsl::time_deleted.is_null())
+                                    .set(volume_dsl::data.eq(rop_volume_data))
+                                    .execute(conn)?;
+                            if num_updated != 1 {
+                                return Err(TxnError::CustomError(
+                                    RemoveReadOnlyParentError::UnexpectedDatabaseState,
+                                ));
+                            }
+                            Ok(true)
+                        }
+                    }
+                    _ => {
```

Comment:
nit, and up to you: I like to spell out the match arms rather than using a catch-all so that if someone adds a new variant then we'll be forced to think about how to handle it here.  Is that worth doing here?

Reply:
Done